DragaliaAPI/Features/Chara/CharaController.cs

```diff
@@ -693,102 +684,88 @@ await storyRepository.Stories
                 unlockedStories.Add(charaStories[nextStoryunlockIndex]);
             }
 
+            // they smoked some shit
+            if (isOmnicite)
+                continue;
+
             bool isOnlyUsingGrowMaterial =
                 charaData.GrowMaterialOnlyStartDate <= time
                 && charaData.GrowMaterialOnlyEndDate >= time;
 
-            // they smoked some shit
-            switch (upgradeMaterialType)
+            if (isOnlyUsingGrowMaterial || useGrowMaterial)
```

Comment:
What causes request.is_use_grow_material to be set? I would assume that also encompasses the logic about whether it's active?

Reply:
the "only" period is used for events where you can only upgrade the chara with convictions, and is_use_grow_material should be used when upgrading the chara with convictions after that point to my knowledge